default.py

```diff
@@ -350,6 +357,43 @@ def select_version(self, game_versions):
         else:
             return None
 
+    def has_inputstream_adaptive(self):
+        """Checks if InputStream Adaptive is installed and enabled."""
+        payload = {
+            'jsonrpc': '2.0',
+            'id': 1,
+            'method': 'Addons.GetAddonDetails',
+            'params': {
+                'addonid': 'inputstream.adaptive',
+                'properties': ['enabled']
+            }
+        }
+        response = xbmc.executeJSONRPC(json.dumps(payload))
+        data = json.loads(response)
+        if 'error' not in data and data['result']['addon']['enabled']:
+            addon_log('InputStream Adaptive is installed and enabled.')
+            return True
+        else:
+            addon_log('InputStream Adaptive is not installed and/or enabled.')
+            addon.setSetting('use_inputstream_adaptive', 'false')  # reset setting
+            return False
+
+    def select_stream_url(self, streams):
+        """Determine which stream URL to use from the dict."""
+        if streams:
+            if not addon.getSetting('use_inputstream_adaptive') == 'true' or not self.has_inputstream_adaptive:
```

Comment:
I recommend flipping the logic here. It's confusing that the "else" is the positive of either of the two negatives. Also, in the end, I expect Adaptive Streaming to be the common path, and that should be first.

Something like:

```
if addon.getSetting('use_inputstream_adaptive') == 'true' and self.has_inputstream_adaptive:
  stream_url = streams['manifest_url']
else:
  stuff
```